staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmapleaselock.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcelock
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ConfigMapLeaseLock is used for migration between the configmap and the lease
+type ConfigMapLeaseLock struct {
+	ConfigMapLock *ConfigMapLock
+	LeaseLock     *LeaseLock
+}
+
+// Get returns the older election record of the configmap and the lease
+func (cmll *ConfigMapLeaseLock) Get() (*LeaderElectionRecord, error) {
+	configMapLock, err := cmll.ConfigMapLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	leaseLock, err := cmll.LeaseLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	// check if the older resource lock is released
```

Comment:
RenewTime is not meaningful as it is not a locally collected timestamp. See:

https://github.com/kubernetes/kubernetes/blob/f2c82f49e8df79bd52c7a9c74b310e2ad7805eb9/staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go#L22-L28

Reply:
We want to panic and restart the component while losing any of two lock. And I don't want to break the interface. Any better implement suggestion for here?